### ISSUE TITLE
[SW-802] Ship Sparkling Water with Spark-friendly h2o-genmodel.jar

### ISSUE THREAD

**Issue by Tagar** [h2oai/sparkling-water#671]
https://0xdata.atlassian.net/browse/SW-802

Please consider adding `h2o-genmodel.jar` along with Sparkling Water releases.

Current `h2o-genmodel.jar` has a number of library version conflicts with Spark. 

We'd like to score mojo through Spark executors. 

We can't use other methods to score (like SW frame scoring) because how much data we have. For example, one of the dataframes we would like to score is 64m records x 6300 columns (90% average variables sparsity). In a compact our propitiatory sparse format layout (varname=varvalue encoding), it takes 0.8 Tb on disk. It would be a several Tb h2o frame in h2o dense serialization (actually we were never able to materialize this as a frame nor normally have hardware resources to do that for each Spark session).

The idea is that we would use `mapPartitions()` Spark call to score records using a mojo model. That mapPartitions's function would for each partition:

- initialize `EasyPredictModelWrapper()` etc just once per partitions
- and then score each record (we'd read directly from our internal sparse propitiatory format - we already have this code ready) using calls like `model.predictBinomial`.. 

The only problem now is that `h2o-genmodel.jar` has inclusive set of Java libraries, some of which conflict with what Spark or Hadoop ships with.

As I explained in [SW-802](https://0xdata.atlassian.net/browse/SW-802):

Would it be possible to build a separate, slim version of `h2o-genmodel.jar` that has no libraries that are known Spark relies on too?

Other solution I guess would build a unified version of `h2o-genmodel.jar` that has all those conflicting libraries shaded.

Thanks

**Comment by NidhiMehta** [comment]
For such support requests we would suggest to kindly consider becoming H2O enterprise customer.
Thanks.

**Comment by jakubhava** [comment]
@Tagar Mojo predictions are already also supported in parallel manner on each Spark executor. Please see H2OMojoModel class in Scala and equivalent in Python.

**Comment by Tagar** [comment]
thanks @jakubhava - I tried to explain above why we can't use that API call - this would require putting whole dataset to score in memory and this OOMs on us because of the amount of data we have or some other unrelated h2o issues on memory management. That's why we wanted to use `h2o-genmodel` methods directly, - read from our sparse propriatory format and score record by record but with high parallelism through spark executors, using `mapPartitions` call.

thanks @tomkraljevic - I checked this yesterday and this was beyond my Java knowledge level how to do this. Since implementing this was very time-critical, we had to reacrhictecture and use Hadoop Streaming - map-only job that reads our file to score, pipes data through our custom scoring mappers (that do use `h2o-genmodel` ). I think @michalkurka is working now on [PUBDEV-5497](https://0xdata.atlassian.net/browse/PUBDEV-5497) so we can try scoring on executors again once that is resolved.

Thanks all!
